app/workers/import_worker.rb

```diff
@@ -0,0 +1,15 @@
+class ImportWorker
+  include Sidekiq::Worker
+
+  def perform(device, json_file)
```

Comment:
I guess becomes more compact with sidekiq and processing file directly.

How about further decreasing methods size?)
For example,
```ruby
def file_array(json_file)
  JSON.parse(json_file)
end

def perform(device, json_file)
  Checkin.transaction do
    file_array(json_file).drop(1).each do |row|
      checkin_create_or_update_from_row!(row)
    end
  end
end

def checkin_create_or_update_from_row!(row)
  attrs = attributes_from_row(row)
  checkin = Checkin.find_by_id(attrs.delete(:id)) || Checkin.new
  checkin.assign_attributes(attrs)
  checkin.save!
end

def attributes_from_row(row)
  attr = row.split(";")
  { lat: attr[1], lng: attr[2], created_at: attr[3], fogged: attr[11], device_id: device }
end
```